### PR TITLE
Source Files: Move to Headers

### PR DESCRIPTION
**Pull request by ax3l**
- not the cleanest way to keep them in `include/` too, but better then accumulating all in the root path
- CMakeLists: Compile host code separately without nvcc "code splitting" and link it statically

### CODE REVIEW

File: src/picongpu/include/plugins/common/particlePatches.cpp [hunk 1]
```diff
@@ -18,7 +18,7 @@
  * If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "include/plugins/common/particlePatches.hpp"
+#include "particlePatches.hpp"
```
Comment (psychocoderHPC) [inline src/picongpu/include/plugins/common/particlePatches.cpp]: please use relative path from the root `include`. In this case `plugins/common/particlePatches.hpp`

File: src/picongpu/CMakeLists.txt [hunk 2]
```diff
@@ -324,17 +324,19 @@ include_directories(BEFORE ${PIC_EXTENSION_PATH}/include)
...
+    ${SRCFILES}
+)
 
 cuda_add_executable(picongpu
     ${CUDASRCFILES}
-    ${SRCFILES}
 )
 
-target_link_libraries(picongpu ${LIBS} m)
+target_link_libraries(picongpu ${LIBS} picongpu-hostonly m)
```
Comment (psychocoderHPC) [inline src/picongpu/CMakeLists.txt]: We should check if the static library is installed with `make install`

File: src/picongpu/include/plugins/common/stringHelpers.cpp [hunk 3]
```diff
@@ -18,7 +18,7 @@
  * If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "include/plugins/common/stringHelpers.hpp"
+#include "stringHelpers.hpp"
```
Comment (psychocoderHPC) [inline src/picongpu/include/plugins/common/stringHelpers.cpp]: please use relative path from the root `inlcude`

File: src/picongpu/include/plugins/hdf5/openPMD/patchReader.cpp [hunk 4]
```diff
@@ -20,7 +20,7 @@
 
 #if( ENABLE_HDF5 == 1 )
 
-#  include "include/plugins/hdf5/openPMD/patchReader.hpp"
+#  include "patchReader.hpp"
```
Comment (psychocoderHPC) [inline src/picongpu/include/plugins/hdf5/openPMD/patchReader.cpp]: please use relative path from the root `inlcude`

File: src/picongpu/CMakeLists.txt (hunk 2 above)
Comment (ax3l) [inline src/picongpu/CMakeLists.txt]: tested and it's not (since it is not part of our install targets)

**Pull request merged**